python/pyarrow/parquet/core.py

```diff
@@ -1327,17 +1327,15 @@ def __init__(self, path_or_paths, filesystem=None, schema=None, *, filters=None,
                     except ValueError:
                         filesystem = LocalFileSystem(use_mmap=memory_map)
                 finfo = filesystem.get_file_info(path_or_paths)
-                if finfo.is_file:
-                    single_file = path_or_paths
                 if finfo.type == FileType.Directory:
                     self._base_dir = path_or_paths
             else:
-                single_file = path_or_paths
+                buffer_reader = path_or_paths
 
         parquet_format = ds.ParquetFileFormat(**read_options)
 
-        if single_file is not None:
-            fragment = parquet_format.make_fragment(single_file, filesystem)
+        if buffer_reader is not None:
+            fragment = parquet_format.make_fragment(buffer_reader, filesystem)
```

Comment:
Can you please remove these unrelated changes?

Reply:
+1

Reply:
Changed back to `single_file`.